js/validation/validationSignIn.js

```diff
@@ -6,49 +6,39 @@ const validationSignIn = () => {
     '.error-message__password'
   )
 
-  if (email.value === '') {
-    email.classList.add('form__input_error')
-    errorMessageEmail.innerHTML = 'Please enter your email.'
-    errorMessageEmail.classList.add('form__text_error')
-    return false
-  } else {
-    email.classList.remove('form__input_error')
-    errorMessageEmail.innerHTML = ''
+  const ifFieldValid = (target, message, reg) => {
+    if (!target.value) {
+      target.classList.add('form__input_error')
+      message.innerHTML = `Please enter your ${target.type}.`
+      message.classList.add('form__text_error')
+
+      return false
+    } else {
+      target.classList.remove('form__input_error')
+      message.innerHTML = ''
+    }
+
+    if (!reg(target.value)) {
+      target.classList.add('form__input_error')
+      message.innerHTML = `Please enter a valid ${target.type}.`
+      message.classList.add('form__text_error')
+
+      return false
+    } else {
+      target.classList.remove('form__input_error')
+      message.innerHTML = ''
+    }
```

Comment:
You can move your logic into one if (!target.value && !reg(target.value)) also these two can be moved into variable: const isFieldValidCheck = target.value && reg(target.value) and in your if (!isFieldValidCheck) do the logic

      target.classList.add('form__input_error')
      message.innerHTML = `Please enter a valid ${target.type}.`
      message.classList.add('form__text_error')